server/channels/app/preference.go

```diff
@@ -94,10 +96,22 @@ func (a *App) UpdatePreferences(userID string, preferences model.Preferences) *m
 	message.Add("preferences", string(prefsJSON))
 	a.Publish(message)
 
+	pluginContext := pluginContext(c)
+	a.Srv().Go(func() {
+		a.ch.RunMultiHook(func(hooks plugin.Hooks) bool {
+			prefPointers := make([]*model.Preference, len(preferences))
```

Comment:
I wasn't sure if I should comment on this in the code, but I made sure that the plugin hook used a slice of pointers instead of a `model.Preferences` which is a `[]model.Preference` under the hood. I forget why I named the type instead of just using an array, but I wanted to fix the mistake I made previously of inconsistently using non-pointers for preferences which I thought was a good idea at the time.

Reply:
For the plugin API, it's mostly stylistic since they get sent over RPC the same way anyway. When I was working on the early preferences code, I was fairly new to Go and thought pointers were bad, so I avoided using them for preferences. I think I was either trying to avoid using the heap (which I'm pretty sure they still do in a slice) or if it's just because I hated using them in C++ in school.

Anyway, I wanted to fix that here since it's inconsistent with methods for other objects. I just realized though that the existing plugin API methods like `PluginAPI.GetPreferencesForUser` all use `Preference` instead of `*Preference` though, so this is now inconsistent with the rest of the plugin API.

Any thoughts on if it'd be better to be more consistent with the rest of the plugin API or more with the other preference-related APIs? I feel like I've spent too long thinking about this and confused myself 😛

Reply:
Yeah, I don't think passing a slice of values vs a slice of pointers to values in this instance makes any practical difference performance wise since everything needs to be serialized anyway. I'd probably keep it simple and consistent with the plugin API which is just more challenging to break. Nothing wrong with `[]model.Preference` that I can see.